### PR TITLE
Basic support added for the use of dates.

### PR DESCRIPTION
**Pull request by jhkuperus**
Current implementation wraps a jodatime LocalDate and supports implicit conversions from String, java Date and jodatime LocalDate.

Tests are included to demonstrate support for comparative operators in DslCondition.

### CODE REVIEW

File: engine/src/main/scala/org/scalarules/dsl/nl/grammar/date/DslDatumImplicits.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  abstract class ToDslDate(internal: LocalDate) {
+    /** Builds DslDate from the provided value. */
+    def datum: Datum = DslDate(internal)
+
+  }
+
+  lazy val dtf = DateTimeFormat.forPattern("dd-MM-yyyy")
+
+  implicit class JodaLocalTimeToDslDate(external: LocalDate) extends ToDslDate(external)
```
Comment (NRBPerdijk) [inline engine/src/main/scala/org/scalarules/dsl/nl/grammar/date/DslDatumImplicits.scala]: You mean: JodaLocalDateToDslDate

Comment (jhkuperus) [inline engine/src/main/scala/org/scalarules/dsl/nl/grammar/date/DslDatumImplicits.scala]: Yep, got distracted by the JodaTime name ;) Fixed it

File: engine/src/test/scala/org/scalarules/dsl/core/DatumTestsBerekening.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package org.scalarules.dsl.core
+
+import org.scalarules.dsl.core.DatumTestGlossary._
+import org.scalarules.dsl.nl.grammar._
+
+class DatumTestsBerekening extends Berekening (
+
+  Gegeven (DatumA < "01-01-2015".datum)
+    Bereken KleinerDan is "success"
```
Comment (NRBPerdijk) [inline engine/src/test/scala/org/scalarules/dsl/core/DatumTestsBerekening.scala]: I understand the use of KleinerDan and co as the equivalent of the mathematical comparator, but it just looks weird in Dutch. How about using "EerderDan" etc? We are writing a DSL afterall. :)

Comment (jhkuperus) [inline engine/src/test/scala/org/scalarules/dsl/core/DatumTestsBerekening.scala]: Do you mean the `<` operator, or the name of the facts in the tests?

The operators are the ones already supported by the DSL and they will remain so. Although I do suggest we add specific date-related words to the DSL as 'aliases' for the mathematical operators.

The names of the facts, I'll concede and rename those to more appropriate names for date related handling. 👌

File: engine/src/main/scala/org/scalarules/dsl/core/grammar/date/DslDate.scala [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.joda.time.LocalDate
+
+/**
+  * Provides the DSL with a Date type. It supports the Ordered trait, which means it can be used
+  * in DslConditions with the comparative operators.
+  *
+  * @param internal internal representation of the date.
+  */
+case class DslDate private[date](internal: LocalDate) extends Ordered[DslDate] {
```
Comment (vzorge) [inline engine/src/main/scala/org/scalarules/dsl/core/grammar/date/DslDate.scala]: I would actually consider calling the class 'LocalDate' and perhaps just place it in the org.scalarules.dsl.core package or one deeper (add .date or .time).
The reason for that is that it reads more intuitive and after all, both java 8 and JodaTime both call the class LocalDate as well, so there can be no name confusion.

For naming, in Scala you can give an Alias, so we can just import JodaTime date as JodaLocalDate

**Comment by jhkuperus** [comment]
Okay, I like this positioning. The `dsl.core` now has a `temporal` package, in which we can add other temporal types as well. The `dsl.nl.grammar` has the `Datum` alias and accompanying implicits inside the `datum` package.

**Comment by vzorge** [comment]
Looks good. Approved!

**Pull request merged**